src/ncp/ncp_base.cpp

```diff
@@ -2584,6 +2584,51 @@ otError NcpBase::DecodeLinkMetrics(otLinkMetrics *aMetrics, bool aAllowPduCount)
 }
 #endif
 
+#if OPENTHREAD_PLATFORM_CONFIG_POWER_CALIBRATION_ENABLE
+template <> otError NcpBase::HandlePropertySet<SPINEL_PROP_RADIO_CHAN_TARGET_POWER>(void)
+{
+    otError error;
+    uint8_t channel;
+    int16_t targetPower;
+
+    SuccessOrExit(error = mDecoder.ReadUint8(channel));
+    SuccessOrExit(error = mDecoder.ReadInt16(targetPower));
+    error = otPlatRadioSetChannelTargetPower(mInstance, channel, targetPower);
+
+exit:
+    return error;
+}
+
+template <> otError NcpBase::HandlePropertyInsert<SPINEL_PROP_RADIO_CALIBRATED_POWER>(void)
+{
+    otError           error;
+    uint8_t           channel     = 0;
+    int16_t           actualPower = 0;
+    otRawPowerSetting rawPowerSetting;
+    const uint8_t *   dataPtr;
+    uint16_t          dataLen;
+
+    SuccessOrExit(error = mDecoder.ReadUint8(channel));
+    SuccessOrExit(error = mDecoder.ReadInt16(actualPower));
+
+    SuccessOrExit(error = mDecoder.ReadDataWithLen(dataPtr, dataLen));
```

Comment:
Looks like we are using `d` (DATA_WITH_LENGTH()` when we decode.

Reply:
I have updated the code to use `d` to encode data.